.github/workflows/publish-docs.yml

```diff
@@ -27,10 +27,13 @@ jobs:
           cache: pip
           cache-dependency-path: docs/requirements.txt
 
-      - name: Set up Go ${{ env.GO_VERSION }}
-        uses: actions/setup-go@v3
+      - name: Set up Go
+        uses: actions/setup-go@v5
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version-file: go.mod
+          cache-dependency-path: |
+            go.sum
+            hack/tool/go.sum
```

Comment:
The go toolchain here is solely required to compile k0s_sort, so there's no need to cache this.

This is also the biggest concern that I have about using setup-go's builtin caching. It's made for the simple case that there's a single workflow that installs go once and always compiles the same stuff. Perfectly reasonable for most of the go projects out there. The k0s workflows are different, and compile different parts of the project in different workflows. Yet, setup-go will always use the same cache for all of those. Here, we're not even compiling anything from k0s's source tree.